internal/test/wmcb/wmcb_test.go

```diff
@@ -635,28 +664,43 @@ func testWMCBCluster(t *testing.T) {
 	assert.Lenf(t, winNodes.Items, 1, "expected one node to have node label but found: %v", len(winNodes.Items))
 }
 
-// initializeCNIPluginsPkgInfo populates the fields of cniPlugins
-func (f *wmcbFramework) initializeCniPluginsPkgInfo() error {
-	f.cniPlugins = pkgInfo{}
+//intitializePkgInfo populates the fields for cniPlugins package
+func (c *CniPlugins) initializePkgInfo() error {
+	var cniPluginsPkg CniPlugins
 
-	f.cniPlugins.url = cniPluginsBaseURL + f.TestFramework.LatestCniPluginsVersion + "/cni-plugins-windows-amd64-" +
-		f.TestFramework.LatestCniPluginsVersion + ".tgz"
+	cniPluginsPkg.url = cniPluginsBaseURL + framework.TestFramework.LatestCniPluginsVersion + "/cni-plugins-windows-amd64-" +
+		framework.TestFramework.LatestCniPluginsVersion + ".tgz"
 
-	cniPluginsSHA, err := framework.getLatestCniPluginsSHA()
+	cniPluginsSHA, err := cniPluginsPkg.getSHAValue(cniPluginsPkg.url)
 	if err != nil {
 		return fmt.Errorf("unable to get SHA for CNI Plugins: %v", err)
 	}
-	f.cniPlugins.sha = cniPluginsSHA
+	cniPluginsPkg.sha = cniPluginsSHA
 
-	f.cniPlugins.shaType = "sha512"
-	return nil
-}
+	cniPluginsPkg.shaType = "sha512"
+	c.pkgInfo = cniPluginsPkg.pkgInfo
+	return nil}
 
 // getLatestCniPluginsSHA returns the SHA512 of the latest 0.8.x version of CNI Plugins
-func (f *wmcbFramework) getLatestCniPluginsSHA() (string, error) {
-	latestCniPluginsURL := f.cniPlugins.url
-	latestCniPluginsChecksumFileURL := latestCniPluginsURL + ".sha512"
-	response, err := http.Get(latestCniPluginsChecksumFileURL)
+func (c *CniPlugins) getSHAValue(latestCniPluginsURL string) (string, error) {
+	checksumFileContent, err := c.getSHAFileContent(latestCniPluginsURL)
+	if err!= nil{
+		return "", err
+	}
+	// The checksum file content is in the format "<sha> <filename>". So to get SHA we need to extract only the <sha>
+	// from the file
+	sha512AndFilename := strings.Split(checksumFileContent, " ")
+	if len(sha512AndFilename) < 2 {
+		return "", fmt.Errorf("checksum file content is not in the format:'<sha> <filename>'")
+	}
+	sha512 := sha512AndFilename[0]
+	return sha512, nil
+}
+
+// getSHAFileContent returns the contents of the SHA file for the given url of the package.
+func (f *pkgInfo) getSHAFileContent(url string)(string, error){
+	pkgChecksumFileURL := url + ".sha512"
+	response, err := http.Get(pkgChecksumFileURL)
 	if err != nil {
 		return "", fmt.Errorf("could not fetch CNI Plugins checksum file: %s", err)
```

Comment:
Do we fetch checksum of only CNI plugins here. Do we need to change error message?